js_modules/dagit/packages/core/src/partitions/__tests__/CreatePartitionDialog.test.tsx

```diff
@@ -63,4 +75,33 @@ describe('CreatePartitionDialog', () => {
     userEvent.type(partitionInput, 'validname');
     expect(screen.queryByTestId('warning-icon')).toBeNull();
   });
+
+  it('Shows error state when mutation fails', async () => {
+    const createFixture = buildCreatePartitionMutation({
+      variables: {
+        partitionsDefName: 'testPartitionDef',
+        partitionKey: 'testPartitionName',
+        repositorySelector: {
+          repositoryLocationName: 'testing',
+          repositoryName: 'testing',
+        },
+      },
+      data: buildUnauthorizedError({
+        message: 'test message 123',
+      }),
+    });
+    await act(async () => {
+      render(<Test mocks={[createFixture]} />);
+    });
```

Comment:
You shouldn't need the `await act` when using `MockedProvider` (instead of `TestProvider`). At least, I haven't needed it so far.